ocaml/libs/tracing/tracing.ml

```diff
@@ -279,6 +279,56 @@ module Spans = struct
 
   let spans = Hashtbl.create 100
 
+  let frequency_tbl = Hashtbl.create 100
+
+  let incr_frequency_tbl ~(span : Span.t) =
+    match Hashtbl.find_opt frequency_tbl span.name with
+    | None ->
+        Hashtbl.add frequency_tbl span.name 1
+    | Some v ->
+        Hashtbl.replace frequency_tbl span.name (succ v)
+
+  let decr_frequency_tbl ~(span : Span.t) =
+    match Hashtbl.find_opt frequency_tbl span.name with
+    | None ->
+        ()
+    | Some v ->
+        if pred v = 0 then
+          Hashtbl.remove frequency_tbl span.name
+        else
+          Hashtbl.replace frequency_tbl span.name (pred v)
+
+  let most_frequent n =
+    let lst =
+      frequency_tbl
+      |> Hashtbl.to_seq
+      |> List.of_seq
+      |> List.sort (fun x y -> compare (snd x) (snd y))
+      |> List.rev
+    in
+    let rec first_n_of_lst acc n lst =
```

Comment:
This function is defined in [Xapi_stdext_std.Listext.List](https://github.com/xapi-project/stdext/blob/master/lib/xapi-stdext-std/listext.mli#L22) Please do not redefine it.